src/main/java/com/abdullah/coding/challenge/controllers/BookingController.java

```diff
@@ -0,0 +1,63 @@
+package com.abdullah.coding.challenge.controllers;
+
+import com.abdullah.coding.challenge.Response;
+import com.abdullah.coding.challenge.dto.BookingDTO;
+import com.abdullah.coding.challenge.services.BookingService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("booking")
+public class BookingController {
+    @Autowired
+    private BookingService service;
+
+    @GetMapping()
+    public ResponseEntity<?> getBooking(@RequestParam("bookingId") int bookingId) {
+        Response<?> response = service.getBooking(bookingId);
+        return new ResponseEntity<>(response, response.getStatusCode());
+    }
+
+    @PostMapping("/create")
+    public ResponseEntity<?> createBooking(@RequestBody BookingDTO bookingDTO) {
+        Response<?> response = service.createBooking(bookingDTO);
+        return new ResponseEntity<>(response, response.getStatusCode());
+    }
+
+    @PatchMapping("/update")
+    public ResponseEntity<?> updateBooking(@RequestBody BookingDTO bookingDTO, @RequestParam("bookingCode") String bookingCode) {
+        Response<?> response = service.updateBooking(bookingDTO, bookingCode);
+        return new ResponseEntity<>(response, response.getStatusCode());
+    }
+
+    @PatchMapping("/cancel")
+    public ResponseEntity<?> cancelBooking(@RequestParam("bookingCode") String bookingCode, @RequestParam("customerId") int customerId) {
```

Comment:
**How this API is different from the management cancellation API?**
- It's not and that's by design. Booking Controller is often used by a customer. If a customer wants to cancel their booking, I created this route to isolate it from any other cancellation API so that in the future I may add certain features that only cater to a customer's booking cancellation.